docs/proposal/layer4_authorization.md

```diff
@@ -56,5 +54,32 @@ struct {
    - 3.2: If the matched record shows `value=deny`, it alters the message flag, sends an RST message to the server, clears the corresponding `auth_map` record. If no record is matched, implying authorization is allowed, the message is passed through.
 4. **client retry**: The client attempts to send another message, but because the server has closed the connection, the client receives a "reset by peer" signal and subsequently closes its own channel.
 
+### Xdp authentication
+
+#### Design detail
+
+![l4_authz_xdp](pics/kmesh_l4_authorization_xdp.svg#pic_center)
+
+#### Map definition
+
+map_of_wl_policy: records the policies that are configured for the workload.
 
+map_of_authz: records the authz rules of policies.
 
+```.c
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(key_size, sizeof(__u32));
+    __uint(value_size, sizeof(wl_policies_v));
+    __uint(map_flags, BPF_F_NO_PREALLOC);
+    __uint(max_entries, MAP_SIZE_OF_AUTH_POLICY);
+}    SEC(".maps");
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(key_size, sizeof(__u32));
+    __uint(value_size, sizeof(Istio__Security__Authorization));
+    __uint(map_flags, BPF_F_NO_PREALLOC);
+    __uint(max_entries, MAP_SIZE_OF_AUTH_POLICY);
+} map_of_authz SEC(".maps");
+```
```

Comment:
Why xdp program return XDP_DROP? 
This causes packet loss and timeout errors on the client.

Reply:
Any other good suggestion?

Reply:
> Any other good suggestion?

There doesn't seem to be a particularly good solution to replace the timeout here.
XDP is a packet-base handler, you can only interfere with the connection by modifying the packet state.
But the connection status can reflect very little information.
Only one RST is returned to the peer end when the connection is terminated. The peer end, however, does not know the cause of the RST.

Reply:
Yeah, how can we make it rest fast instead of waiting for timeout now